src/python/pants/backend/python/rules/coverage/plugin.py

```diff
@@ -0,0 +1,97 @@
+# coding=utf-8
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import json
+import os
+
+from coverage import CoveragePlugin, FileTracer
+from coverage.config import DEFAULT_PARTIAL, DEFAULT_PARTIAL_ALWAYS
+from coverage.misc import join_regex
+from coverage.parser import PythonParser
+from coverage.python import PythonFileReporter
+
+
+# NB: This file must keep Python 2 support because it is a resource that may be run with Python 2.
+
+
+
+
+class SimpleFileTracer(FileTracer):
+  def __init__(self, filename):
+    super(SimpleFileTracer, self).__init__()
+    self._filename = filename
+
+  def source_filename(self):
+    return self._filename
+
+
+class SimpleFileReporter(PythonFileReporter):
+  def __init__(self, morf, relpath):
+    super(SimpleFileReporter, self).__init__(morf, coverage=None)
+    self._relpath = relpath
+
+  def relative_filename(self):
+    return self._relpath
+
+  # TODO(John Sirois): Kill the workaround overrides below if there is a useable upstream
+  # resolution to:
+  #   https://bitbucket.org/ned/coveragepy/issues/646/modifying-coverage-reporting-for-python
+
+  @property
+  def parser(self):
+    if self._parser is None:
+      self._parser = PythonParser(filename=self.filename)
+      self._parser.parse_source()
+    return self._parser
+
+  def no_branch_lines(self):
+    return self.parser.lines_matching(join_regex(DEFAULT_PARTIAL[:]),
+                                      join_regex(DEFAULT_PARTIAL_ALWAYS[:]))
+
+
+class ChrootRemappingPlugin(CoveragePlugin):
```

Comment:
As far as I can tell, this doesn't have anything to do with chroots - it re-adds source-roots back to things which have been source-root-stripped? Maybe rename things accordingly?